src/agent/multitype_queue/src/multitype_queue.cpp

```diff
@@ -185,15 +185,29 @@ boost::asio::awaitable<std::vector<Message>> MultiTypeQueue::getNextNAwaitable(M
 }
 
 std::vector<Message> MultiTypeQueue::getNextN(MessageType type,
-                                              int messageQuantity,
+                                              std::variant<const int, const size_t> messageQuantity,
                                               const std::string moduleName,
                                               const std::string moduleType)
 {
     std::vector<Message> result;
     if (m_mapMessageTypeName.contains(type))
     {
-        auto arrayData =
-            m_persistenceDest->RetrieveMultiple(messageQuantity, m_mapMessageTypeName.at(type), moduleName, moduleType);
+        nlohmann::json arrayData;
+        if (std::holds_alternative<const int>(messageQuantity))
+        {
+            arrayData = m_persistenceDest->RetrieveMultiple(
+                std::get<const int>(messageQuantity), m_mapMessageTypeName.at(type), moduleName, moduleType);
+        }
+        else if (std::holds_alternative<const size_t>(messageQuantity))
+        {
+            arrayData = m_persistenceDest->RetrieveBySize(
+                std::get<const size_t>(messageQuantity), m_mapMessageTypeName.at(type), moduleName, moduleType);
+        }
```

Comment:
Despite having warnings enabled that would tell us about implicit conversions between ints and size_t, I don't think that using these types is enough to safely make this distinction. I would either have two separate functions that call the appropiate RetrieveMultiple or RetrieveBySize or explicitly create types that cannot be converted between each other.